packages/itwin/tree-widget/src/components/TreeWidgetUiItemsProvider.tsx

```diff
@@ -26,64 +48,32 @@ export class TreeWidgetUiItemsProvider implements UiItemsProvider {
     section?: StagePanelSection
   ): ReadonlyArray<Widget> {
-    const widgets: Widget[] = [];
     const preferredLocation = this._treeWidgetOptions?.defaultPanelLocation ?? StagePanelLocation.Right;
     const preferredPanelSection = this._treeWidgetOptions?.defaultPanelSection ?? StagePanelSection.Start;
-    if (
-      location === preferredLocation &&
-      section === preferredPanelSection &&
-      stageUsage === StageUsage.General
-    ) {
-      const trees: SelectableContentDefinition[] = [];
-
-      if (!this._treeWidgetOptions?.hideTrees?.modelsTree) {
-        trees.push({
-          label: TreeWidget.translate("models"),
-          id: ModelsTreeId,
-          render: () => (
-            <ModelsTreeComponent
-              enableElementsClassGrouping={
-                this._treeWidgetOptions?.enableElementsClassGrouping
-                  ? ClassGroupingOption.YesWithCounts
-                  : ClassGroupingOption.No
-              }
-              {...this._treeWidgetOptions?.modelsTreeProps}
-            />
-          ),
-        });
-      }
 
-      if (!this._treeWidgetOptions?.hideTrees?.categoriesTree) {
-        trees.push({
-          label: TreeWidget.translate("categories"),
-          id: CategoriesTreeId,
-          render: () => (
-            <CategoriesTreeComponent {...this._treeWidgetOptions?.categoriesTreeProps} />
-          ),
-        });
-      }
-
-      if (this._treeWidgetOptions?.additionalTrees) {
-        trees.push(...this._treeWidgetOptions.additionalTrees);
-      }
-
-      if (this._treeWidgetOptions?.defaultTreeId && trees.length !== 0) {
-        // Adding the defaultTree to first index
-        const { defaultTreeId } = this._treeWidgetOptions;
-        const extractedDefaultTree = trees.filter((tree) => tree.id === defaultTreeId)[0];
-        const index = trees.indexOf(extractedDefaultTree);
-        trees.unshift(trees.splice(index, 1)[0]);
-      }
-
-      widgets.push({
-        id: TreeWidgetId,
-        label: TreeWidget.translate("treeview"),
-        content: <TreeWidgetComponent trees={trees} />,
-        icon: "icon-hierarchy-tree",
-        priority: this._treeWidgetOptions?.defaultTreeWidgetPriority,
-      });
+    if (location !== preferredLocation || section !== preferredPanelSection || stageUsage !== StageUsage.General) {
+      return [];
     }
 
-    return widgets;
+    const trees: TreeDefinition[] = this._treeWidgetOptions?.trees ?? [
+      {
+        id: ModelsTreeComponent.id,
+        getLabel: ModelsTreeComponent.getLabel,
+        render: () => <ModelsTreeComponent />,
+      },
+      {
+        id: CategoriesTreeComponent.id,
+        getLabel: CategoriesTreeComponent.getLabel,
+        render: () => <CategoriesTreeComponent />,
+      },
+    ];
+
+    return [{
+      id: TreeWidgetId,
+      label: TreeWidget.translate("treeview"),
+      content: <TreeWidgetComponent trees={trees} />,
+      icon: "icon-hierarchy-tree",
```

Comment:
is there an icon from the icons pkg we can use instead of from the svg-webfont icons

Reply:
Change to icon from `@itwin/itwinui-icons-react` package